src/hotspot/share/nmt/virtualMemoryTracker.cpp

```diff
@@ -47,545 +46,213 @@ void VirtualMemory::update_peak(size_t size) {
-  if (node != nullptr) {
-    CommittedMemoryRegion* rgn = node->data();
-
-    if (is_mergeable_with(rgn, addr, size, stack)) {
-      rgn->expand_region(addr, size);
-      return true;
-    }
-  }
-
-  return false;
+void VirtualMemoryTracker::set_reserved_region_tag(address addr, MemTag mem_tag) {
+    ReservedMemoryRegion rgn = tree()->find_reserved_region(addr);
+    VMATree::SummaryDiff diff = tree()->set_tag((VMATree::position) addr, rgn.size(), mem_tag);
+    apply_summary_diff(diff);
 }
 
-static bool try_merge_with(LinkedListNode<CommittedMemoryRegion>* node, LinkedListNode<CommittedMemoryRegion>* other) {
-  if (other == nullptr) {
-    return false;
-  }
-
-  CommittedMemoryRegion* rgn = other->data();
-  return try_merge_with(node, rgn->base(), rgn->size(), *rgn->call_stack());
+void VirtualMemoryTracker::Instance::apply_summary_diff(VMATree::SummaryDiff diff) {
+  assert(_tracker != nullptr, "Sanity check");
+  _tracker->apply_summary_diff(diff);
 }
 
-bool ReservedMemoryRegion::add_committed_region(address addr, size_t size, const NativeCallStack& stack) {
-  assert(addr != nullptr, "Invalid address");
-  assert(size > 0, "Invalid size");
-  assert(contain_region(addr, size), "Not contain this region");
-
-  // Find the region that fully precedes the [addr, addr + size) region.
-  LinkedListNode<CommittedMemoryRegion>* prev = find_preceding_node_from(_committed_regions.head(), addr);
-  LinkedListNode<CommittedMemoryRegion>* next = (prev != nullptr ? prev->next() : _committed_regions.head());
-
-  if (next != nullptr) {
-    // Ignore request if region already exists.
-    if (is_same_as(next->data(), addr, size, stack)) {
-      return true;
-    }
-
-    // The new region is after prev, and either overlaps with the
-    // next region (and maybe more regions), or overlaps with no region.
-    if (next->data()->overlap_region(addr, size)) {
-      // Remove _all_ overlapping regions, and parts of regions,
-      // in preparation for the addition of this new region.
-      remove_uncommitted_region(addr, size);
-
-      // The remove could have split a region into two and created a
-      // new prev region. Need to reset the prev and next pointers.
-      prev = find_preceding_node_from((prev != nullptr ? prev : _committed_regions.head()), addr);
-      next = (prev != nullptr ? prev->next() : _committed_regions.head());
+void VirtualMemoryTracker::apply_summary_diff(VMATree::SummaryDiff diff) {
+  VMATree::SingleDiff::delta reserve_delta, commit_delta;
+  size_t reserved, committed;
+  MemTag tag = mtNone;
+  auto print_err = [&](const char* str) {
+    return;
+    // log_error(nmt)("summary mismatch, at %s, for %s,"
+    //                " diff-reserved:  %ld"
+    //                " diff-committed: %ld"
+    //                " vms-reserved: %zu"
+    //                " vms-committed: %zu",
+    //                str, NMTUtil::tag_to_name(tag), (long)reserve_delta, (long)commit_delta, reserved, committed);
+  };
```

Comment:
Any plan regarding this?

Reply:
Will be commented out after https://github.com/openjdk/jdk/pull/23771.
Until then, the error messages will pollute the output and corrupt the jdk-image.